File: 250226/비오는 날/rainy-day.cpp
#include <iostream>
#include <string>
#include <tuple>
#include <vector>
#include <algorithm>
using namespace std;

int n;
string date[100];
string day[100];
string weather[100];

int main() {
    cin >> n;
    for (int i = 0; i < n; i++) {
        cin >> date[i] >> day[i] >> weather[i];
    }

    vector<string> rainy_dates;
    vector<tuple<string, string, string>> v;
    for(int i = 0;i < n;i++){
        if(weather[i] == "Rain"){
            rainy_dates.push_back(date[i]);
            v.push_back(make_tuple(date[i], day[i], weather[i]));
        }
    }

    sort(rainy_dates.begin(), rainy_dates.end());
    for(int i = 0; i < v.size();i++){
        if(rainy_dates[0] == get<0>(v[i])){
            string a, b, c;
            tie(a, b, c) = v[i];
            cout << a << " " << b << " " << c << "\n";
        }
    }
    return 0;
}

File: 250226/비오는 날/rainy-day.py
n = int(input())
dates = []
days = []
weathers = []

class Weather_Data:
    def __init__(self, date, day, weather):
        self.date_ = date
        self.day_ = day
        self.weather_ = weather

weather_datas = []
for i in range (n):
    date, day, weather = tuple(input().split())
    if weather == "Rain":
        dates.append(date)
        weather_datas.append(Weather_Data(date, day, weather))

for i in range (len(weather_datas)):
    if weather_datas[i].date_ == sorted(dates)[0]:
        print(weather_datas[i].date_, weather_datas[i].day_, weather_datas[i].weather_)
